### ISSUE TITLE
Second deploy fail on Lambda even after disconnecting from WebSocket connection

### ISSUE THREAD

**Issue by abbaseya** [localstack/localstack#4626]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The latest LocalStack has to get restarted each time I needed to post an update on Lambda, that wasn't an issue before, but now I'm getting a `KeyError` instead.

### Expected Behavior

Lambda should be updated when re-deployed using Serverless

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

```yml
version: '3.9'

services:
    localstack:
        container_name: localstack
        image: localstack/localstack-full
        ports:
            - 443:443
            - 4510-4530:4510-4530
            - 4566:4566
            - 4571:4571
        environment:
            - LOCALSTACK_API_KEY=${LOCALSTACK_LICENSE}
            - USE_LIGHT_IMAGE=0
            - IMAGE_NAME=localstack/localstack-full
            - MAIN_CONTAINER_NAME=localstack
            - SERVICES=cloudformation,cloudfront,apigateway,apigatewayv2,iam,secretsmanager,lambda,s3,sqs,sts,ec2,kafka,elb,elbv2
            - DEFAULT_REGION=us-east-1
            - AWS_ACCESS_KEY_ID=test
            - AWS_SECRET_ACCESS_KEY=test
            - DEBUG=1
        volumes:
            - /var/run/docker.sock:/var/run/docker.sock
        network_mode: bridge
```

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

A test case available at [GitHub](https://github.com/abbaseya/localstack-msk-lambda-test) - test steps:

1. Start LocalStack which deploys and re-deploys a couple of Lambda functions:`./socket.sh`
2. Observe  LocalStack logs around `KeyError`:
```
2021-09-21T14:45:31:ERROR:cloudformation_api: Exception on / [POST]
Traceback (most recent call last):
  File "/opt/code/localstack/.venv/lib/python3.7/site-packages/flask/app.py", line 2070, in wsgi_app
    response = self.full_dispatch_request()
  File "/opt/code/localstack/.venv/lib/python3.7/site-packages/flask/app.py", line 1515, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/opt/code/localstack/.venv/lib/python3.7/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/opt/code/localstack/.venv/lib/python3.7/site-packages/flask/app.py", line 1513, in full_dispatch_request
    rv = self.dispatch_request()
  File "/opt/code/localstack/.venv/lib/python3.7/site-packages/flask/app.py", line 1499, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 978, in handle_request
    result = func(req_params)
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 685, in list_stack_resources
    result = describe_stack_resources(req_params)
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 678, in describe_stack_resources
    for res_id, res_status in stack.resource_states.items()
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 181, in resource_states
    self._set_resource_status_details(resource_id)
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 161, in _set_resource_status_details
    resource = self.resources[resource_id]
KeyError: 'WsOnMessageLambdaVersionL1kXQQ2FOSlubDPB5KYey3uVLSeLYW7UYQS7Am6g'
```

### Environment

```markdown
- OS: macOS 11.5.2
- LocalStack: latest
- AWS CLI: 2.2.35
- Serverless: 2.57.0
- WSCAT: 5.0.0
```

**Comment by whummer** [comment]
Thanks for reporting @abbaseya . This should be fixed via #4636 - can you please pull the `latest` Docker image and give it another try? Please let us know if the problem persists. Thanks for your help!

**Comment by abbaseya** [comment]
@whummer This is to confirm that the issue has been resolved after pulling the latest LocalStack image. Thank you so much 🙌🏻
